models/user.go

```diff
@@ -321,7 +323,7 @@ func findUser(tx *storage.Connection, query string, args ...interface{}) (*User,
 
 // FindUserByConfirmationToken finds users with the matching confirmation token.
 func FindUserByConfirmationToken(tx *storage.Connection, token string) (*User, error) {
-	user, err := findUser(tx, "confirmation_token = ?", token)
+	user, err := findUser(tx, "confirmation_token = ? and is_sso_user = false", token)
```

Comment:
why do we need to add `is_sso_user = false` here? there's already a unique constraint on `confirmation_token`

Reply:
To be extra cautious that you're not accidentally finding a user that does not support confirmation tokens. For example, someone accidentally flipping the switch to `is_sso_user = true` of an existing user. SSO users can't be "confirmed" as they only are accessible via the SAML ACS endpoint.